src/ai/backend/manager/models/domain.py

```diff
@@ -213,3 +220,64 @@ async def mutate(cls, root, info, name):
             .where(domains.c.name == name)
         )
         return await simple_db_mutate(cls, info.context, query)
+
+
+class PurgeDomain(graphene.Mutation):
+    '''
+    Completely delete domain from DB.
+
+    Domain-bound kernels will also be all deleted.
+    To purge domain, there should be no users and groups in the target domain.
+    '''
+    allowed_roles = (UserRole.SUPERADMIN,)
+
+    class Arguments:
+        name = graphene.String(required=True)
+
+    ok = graphene.Boolean()
+    msg = graphene.String()
+
+    @classmethod
+    async def mutate(cls, root, info, name):
+        from . import users, groups
+        async with info.context['dbpool'].acquire() as conn:
+            query = (
+                sa.select([sa.func.count()])
+                .where(users.c.domain_name == name)
+            )
+            user_count = await conn.scalar(query)
+            assert user_count == 0, 'There are users bound to the domain. Remove users first.'
+            query = (
+                sa.select([sa.func.count()])
+                .where(groups.c.domain_name == name)
+            )
+            group_count = await conn.scalar(query)
+            assert group_count == 0, 'There are groups bound to the domain. Remove groups first.'
+
+            await cls.delete_kernels(conn, name)
+        query = domains.delete().where(domains.c.name == name)
+        return await simple_db_mutate(cls, info.context, query)
+
+    @classmethod
+    async def delete_kernels(
+        cls,
+        conn: SAConnection,
+        domain_name: str,
+    ) -> int:
+        """
+        Delete all kernels run from the target domain.
+
+        :param conn: DB connection
+        :param domain_name: domain's name to delete kernels
+
+        :return: number of deleted rows
+        """
+        from . import kernels
+        query = (
+            kernels.delete()
+            .where(kernels.c.domain_name == domain_name)
```

Comment:
I think we need to block this operation if there are non-terminated kernels in the given domain because the actual containers will become orphans and no longer tracked by Backend.AI, meaning resource leaks.